EB:520 + EB:521: Updates to the species observation map

RMCrean opened the pull request:
This PR covers 2 related jira tasks:

### EB-520: Investigate potential issues with the Leaflet species map (regarding showing the error message even though it displays correctly - see image from dev cluster on green toad)
![Screenshot from 2025-05-05 16-38-11](https://github.com/user-attachments/assets/75413e9c-200b-4464-92c8-8de6c47df35c)
I've made some changes and refactored the `hugo/assets/js/gbif_map.js` here which _should_ mean this is handled better. For one thing on page load this custom error message is set to display:none so refreshing the page should now clear the error, which is does not seem to do in the current version as seen on the dev cluster which is displaying the green toad branch.    

### EB-521: Make showing the species distribution map optional
I've added some if else logic to the Hugo species intro page template `hugo/layouts/species/species_intro.html ` to be able to handle a case where a species does not have a distribution map to show. In such a case a user would delete the key `.Params.initialZoom` in the species `_index.md` file to activate the alternative path of not showing the observation map. To document this change I have made a commit in the partially-automate-new-species-submissions instead of here because this is where most of that work is going on, see commit fe3e210482a0aad597466f5e13d1d2538b0378a0).

Now the page for a species without an observation map would look something like this: 
![Screenshot from 2025-05-05 16-57-50](https://github.com/user-attachments/assets/b2211f69-f405-4499-8eb8-2ffbe05121b5)
 

The design/layout for the species pages without a map could probably be improved but that can perhaps be a different Jira task.

RMCrean committed:
Create distrib_map error message in html not js

Instead will just swap between display:none or block if error.

Also removed a duplicated div in the html.

RMCrean committed:
refactor: better error handling + state management

- Create an global state object to manage the state of the map and its components
- Change error handling strategy to hopefully be more cache resilient
- On each new page load, map error message will be cleared.

RMCrean committed:
Make map tooltip look good in dark mode

RMCrean committed:
Make showing observation map optional

RMCrean committed:
Create layout for species pages without a map

Simply just make the image bigger and centered for now at least.

brinkdp reviewed (commented):
Thanks! It seem to work as desired when loading up the species pages. But if I play around with the zoom levels on the map, I eventually get this error message that does not go away unless I reload the page:

![Screenshot 2025-05-06 at 08 48 21](https://github.com/user-attachments/assets/b75117ae-717d-443c-aba0-bc47dd72fc4d)

It happens on all species for me if I click around in the map long enough. Tested on Chrome and Safari, with and without disabling cache.

RMCrean committed:
Style help popup for light and dark mode

RMCrean committed:
Dont raise tileerror for cancelled map requests

tileerror occurs for a failed GBIF API request.

Previously if a user made several rapid requests, then any in progress requests (for a previous map state)
would be cancelled and this would raise a tileerror
This is not an error, so now behaviour is just to check on load for a tileerror.

This initial check instead guards against a bad taxon ID and/or the GBIF API being down.

(see discussion in PR #98 for details).

RMCrean commented:
@brinkdp, thanks for the report, now fixed!  

This error (`tileerror`) comes from when the browser cancelling an in progress request because the map view changed before the data for all of the previous view was loaded.  See screenshot below for example cancelled requests. 
![Screenshot from 2025-05-06 10-12-00](https://github.com/user-attachments/assets/d0afaf29-477e-43c8-9797-a35de482d5aa)

As this is not an error, we now just check on initial load for a `tileerror`, which could come about if the taxon ID used doesn't exist or the GBIF API is down. If during a session the GBIF API went down we wouldn't catch that but that seems very unlikely, would be resolved after refreshing the page.  

I will merge this PR now so we can test it on your new species branches :chipmunk: :rocket:

RMCrean commented:
See commit 92dcbb3ed4f165a57993e7d9b8e26d5ab42cc3e9 for hopefully the final commit related to this little saga. The comments above the function for `hugo/assets/js/gbif_map.js` have been updated according to this change.

Essentially after this PR was merged, we still weren't handling 204 No Content responses correctly. They cause Leaflet to raise a `tileerror` and you can get these even on initial load of the map (as learned from green toad species). This is now handled by the commit referenced above by checking the header content of the response message and seeing if it is 204. If it is 204 we do not raise an error. 

You would also get a 204 if the taxon id provided does not exist so the checks no longer guard against that (you get a 400 if the request is badly formatted which would raise the error). It was decided that checking the taxon id is beyond the scope of this code. A wrong taxon id is easy to spot by just looking at the map and seeing no data after all...  

I will update here if another commit needs to be added, but hopefully this is my last message on this branch :crossed_fingers:
